app/components/InitiateReverifyStampsButton.tsx

```diff
@@ -100,7 +117,7 @@ export const ReverifyStampsModal = ({ isOpen, onClose }: ExpiredStampModalProps)
       }
     });
 
-    await claimCredentials(handleClaimStep, [evmStampClaim, ...stampClaims]);
+    await claimCredentials(handleClaimStep, indicateError, [evmStampClaim, ...stampClaims]);
 
```

Comment:
What is actually required is that after this claiming step, we delete any expired stamps for providers that have not been verified successfully.

For example if we are thinking about the staking stamps, if the user has un-staked, and his stamp expired, after re-verification step (which will now fail), we do not want to show him any more that he has an expired stamp.

Reply:
Yeah I understand the use case. How are you testing this? If the user no longe qualifies for the stamp `await handlePatchStamps(stampPatches);` will delete the stamp if they do not qualify for it. Since it will be filtered out in.

`const stampPatches: StampPatch[] = selectedProviders.map((provider: PROVIDER_ID) => {
            const cred = verifiedCredentials.find((cred: any) => cred.record?.type === provider);
            if (cred) return { provider, credential: cred.credential as VerifiableCredential };
            else return { provider };
          });`